ext_mongodb.php

```diff
@@ -19,6 +19,108 @@ function bsonUnserialize(array $data) : void;
 {
 }
 
+/* == MONITORING ============================================================= */
+namespace MongoDB\Driver\Monitoring;
+
+/* ------ Building blocks for value obejcts */
+abstract class CommandEvent
+{
+	private function __construct()
+	{
+		throw new Exception\RunTimeException("Accessing private constructor");
+	}
+
+	public function getCommandName() : string;
+
+	// do we need this? I don't think so, as you pass in an object
+	public function getContext() : mixed;
+
+	// libmongoc use getHost, but spec says "connectionId".
+	public function getHost() : \MongoDB\Driver\Server;
+
+	// Not adding getHint, as we don't really expose that.
+
+	// technically int64_t. Spec has "Id", and not "ID".
+	public function getOperationId() : integer;
+
+	// technically int64_t
+	public function getRequestId() : integer;
+}
+
+/* This is a separate one, so that one class can implement all three classes without
+ * a duplicate defined interface method */
+abstract class CommandResultEvent extends CommandEvent
+{
+	// spec is unitless, libmongoc does microseconds, Spec suggests "Nanos" as
+	// another alternative. (technically int64_t)
+	public function getDurationMS() : integer;
+}
+
+/* ------ Value Objects ------------------- */
+final class CommandStartedEvent extends CommandEvent
+{
+	// raw command document, with default BSON conversion rules
+	public function getCommand() : object;
+
+	public function getDatabaseName() : string;
+}
+
+final class CommandSucceededEvent extends CommandResultEvent
+{
+	// raw command reply, with default conversion rules
+	public function getReply() : object;
```

Comment:
Perhaps down the line we can consider having Subscribers define their own type map to apply to `getReply()` and `getCommand()`.